.pipelines/cni/lsg/kernel-upgrade-template.yaml

```diff
@@ -0,0 +1,307 @@
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                echo cns version - $(cnsVersion)
+                echo cni version - $(cniVersion)
+
+                echo "Start Integration Tests on Overlay Cluster"
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                kubectl cluster-info
+                kubectl get po -owide -A
+                sudo -E env "PATH=$PATH" make test-integration CNS_VERSION=$(cnsVersion) CNI_VERSION=$(cniVersion) INSTALL_CNS=true INSTALL_AZURE_CNI_OVERLAY=true CNS_IMAGE_REPO=$(CNS_IMAGE_REPO) CNI_IMAGE_REPO=$(CNI_IMAGE_REPO)
+            name: "overlaye2e"
+            displayName: "Overlay Integration"
+            retryCountOnTaskFailure: 2
+      - job: kernel_upgrade
+        displayName: "Kernel Node Upgrade"
+        dependsOn: integration
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                set -ex
+                az extension add --name aks-preview
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+            retryCountOnTaskFailure: 3
+            name: "config"
+            displayName: "Set Kubeconfig"
+          - script: |
+              set -e
+
+              echo "-- Start privileged daemonset --"
+              kubectl get pods -Aowide
+              kubectl apply -f test/integration/manifests/load/privileged-daemonset.yaml
+              sleep 10s
+              kubectl rollout status ds -n kube-system privileged-daemonset
+            retryCountOnTaskFailure: 3
+            name: "priv_daemonset"
+            displayName: "Create Daemonset"
+          - script: |
+              set -e
+
+              kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide
+              privList=`kubectl get pods -n kube-system -l os=linux,app=privileged-daemonset -owide --no-headers | awk '{print $1}'`
+              for pod in $privList; do
+                echo "-- Update Ubuntu Packages --"
+                # Not needed, but ensures that the correct packages exist to perform upgrade
+                kubectl exec -i -n kube-system $pod -- bash -c "apt update && apt-get install software-properties-common -y"
+              done
+            retryCountOnTaskFailure: 3
+            name: "Ubuntu_Setup"
+            displayName: "Ubuntu Setup"
+          - script: |
+              echo "-- Update kernel through daemonset --"
```

Comment:
Will the `set -e` of previous script be applicable here to ?

Reply:
The final checks at the end of the script will do the error capture. I don't want to deal with false negative exit codes coming from kubectl exec.